src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -217,6 +217,253 @@ .maxstack 8
+        }
+
+        [Fact] // PROTOTYPE(semi-auto-props): Add test with semi-colon setter when mixed scenarios are supported.
+        public void TestPropertyIsReadThenAssignedInStructConstructor_ReadOnlyProperty()
+        {
+            var comp = CreateCompilation(@"
+public struct C
+{
+    public static void Main()
+    {
+        var x = new C();
+        System.Console.Write(x.P);
+    }
+
+    public C()
+    {
+        P = 5;
+    }
+
+    public int P { get => field; }
+}
+", options: TestOptions.ReleaseExe.WithSpecificDiagnosticOptions(ReportStructInitializationWarnings));
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+            CompileAndVerify(comp, expectedOutput: "5").VerifyIL("C..ctor", @"
+{
+  // Code size       15 (0xf)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldc.i4.0
+  IL_0002:  stfld      ""int C.<P>k__BackingField""
+  IL_0007:  ldarg.0
+  IL_0008:  ldc.i4.5
+  IL_0009:  stfld      ""int C.<P>k__BackingField""
+  IL_000e:  ret
+}
+").VerifyDiagnostics(
+    // (12,9): warning CS9020: The 'this' object is read before all of its fields have been assigned, causing preceding implicit assignments of 'default' to non-explicitly assigned fields.
+    //         P = 5;
+    Diagnostic(ErrorCode.WRN_UseDefViolationThisSupportedVersion, "P").WithArguments("this").WithLocation(12, 9)
+    );
+            Assert.Empty(comp.GetTypeByMetadataName("C").GetMembers().OfType<FieldSymbol>());
+            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
+        }
+
+        [Fact]
+        public void TestPropertyIsAssignedInStructConstructor_PropertyHasSetter()
+        {
+            var comp = CreateCompilation(@"
+public struct C
+{
+    public static void Main()
+    {
+        var x = new C();
+        System.Console.Write(x.P + "" ""); // 5
+        x.P = 10;
+        System.Console.Write(x.P + "" ""); // 10
+        x = new C();
+        System.Console.Write(x.P); // 5
+    }
+
+    public C()
+    {
+        P = 5;
+    }
+
+    public int P { get => field; set => field = value; }
+}
+", options: TestOptions.ReleaseExe.WithSpecificDiagnosticOptions(ReportStructInitializationWarnings));
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+            CompileAndVerify(comp, expectedOutput: "5 10 5").VerifyIL("C..ctor", @"
+{
+  // Code size       15 (0xf)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldc.i4.0
+  IL_0002:  stfld      ""int C.<P>k__BackingField""
+  IL_0007:  ldarg.0
+  IL_0008:  ldc.i4.5
+  IL_0009:  call       ""void C.P.set""
+  IL_000e:  ret
+}
+").VerifyDiagnostics(
+    // (16,9): warning CS9020: The 'this' object is read before all of its fields have been assigned, causing preceding implicit assignments of 'default' to non-explicitly assigned fields.
```

Comment:
>// (16,9): warning CS9020: The 'this' object is read before all of its fields have been assigned, causing preceding implicit assignments of 'default' to non-explicitly assigned fields. [](http://example.com/codeflow?start=4&length=184)

Perhaps zero initialization of fields created for properties like the one we are testing shouldn't cause the warning. There is no way for a user to initialize the field directly, other than a property initializer. Something to discuss at LDM. A PROTOTYPE comment is good for now.

Reply:
dataflow bug. Okay with adding a PROTOTYPE for now?

Reply:
> dataflow bug. Okay with adding a PROTOTYPE for now?

I do not think there is a bug in flow analysis for this scenario. There is a user-defined setter, we are invoking it, fields should be zero-ed out. The question I had is whether the warning should be reported as well.

Reply:
@AlekseyTs Yes sorry. I was answering thinking it's the other test where the property didn't have a setter.

https://github.com/dotnet/csharplang/issues/6080